logging: Add option to suppress level and timestamp printing in log_output

nordic-krch opened the pull request:
Added flags in log_output module to add severity level and timestamp when message is
formatted to string. Updated existing backends.

Requested by @jhedberg for bluetooth backend he is working on. Same as other log_output flags this flag is also positive (flag must be set to get severity level). Because of that existing backends had to be updated.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
logging: Add option to suppress level printing in log_output

Added flag in log_output module to add severity level when message is
formatted to a string. Updated existing backends.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

jhedberg reviewed (approved):
Looks good to me, except for the minor coding style thing with the extra empty line (see my separate comment).

nordic-krch commented on subsys/logging/log_output.c:
```diff
@@ -166,14 +166,22 @@ static void color_postfix(struct log_msg *msg,
+		total += print_formatted(log_output, "<%s> ", severity[level]);
+	}
+
+
```
removed

carlescufi reviewed (approved):
LGTM

nordic-krch committed:
logging: Add option to suppress timestamp printing in log_output

Added flag in log_output module to add timestamp when message is
formatted to a string. Updated existing backends.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>
